### PR TITLE
Bump deps to 1.5

### PR DESCRIPTION
**Pull request by skonto**
## Proposed Changes

- As per title. This completes the setup for 1.26.
- Split in two commits for easier review.

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -33,6 +32,7 @@ REPLACE_DEPS=(
   "knative.dev/serving=github.com/openshift/knative-serving@${SERVING_VERSION}"
   "knative.dev/pkg=knative.dev/pkg@${KN_VERSION}"
   "knative.dev/hack=knative.dev/hack@${KN_VERSION}"
+  "knative.dev/networking=knative.dev/networking@${KN_VERSION}"
```
Comment (skonto) [inline hack/update-deps.sh:35]: If we don't move there is a conflict with networking versions.

File: hack/update-deps.sh [hunk 2]
```diff
@@ -13,16 +13,15 @@ set -o pipefail
 cd "${ROOT_DIR}"
 
 # This controls the knative release version we track.
-KN_VERSION="release-1.4"
-EVENTING_VERSION="release-v1.4"
+KN_VERSION="release-1.5"
+EVENTING_VERSION="release-v1.5"
 EVENTING_KAFKA_VERSION="release-v1.4"
```
Comment (skonto) [inline hack/update-deps.sh:18]: @pierDipi do we have a plan to remove the deprecated eventing_kafka bits. Should I go on and do it in another PR?

Comment (pierDipi) [reply hack/update-deps.sh:18]: I think we can remove the code dependency now, thanks for your help on this!

**Review by pierDipi — APPROVED** [review]
Thanks! 

If CI is happy, I'm happy with this PR, so
/lgtm

**Comment by skonto** [comment]
As a reminder for future lint updates TODO:

> $ golangci-lint run
knative-operator/cmd/manager/main.go:132:13: G114: Use of net/http serve function that has no support for setting timeouts (gosec)
		if err := http.ListenAndServe(":8080", nil); err != nil && !errors.Is(err, http.ErrServerClosed) {
		          ^
$ /golangci-lint version
golangci-lint has version 1.49.0 built from cc2d97f3 on 2022-08-24T10:24:37Z

We are on 1.45 so it is ok for now.

File: hack/update-deps.sh (hunk 2 above)
Comment (pierDipi) [reply hack/update-deps.sh:18]: We should still keep the test-source-image reference, only the code/vendor dependency can be removed for now.

**Comment by skonto** [comment]
@pierDipi ready for another stamp.

File: hack/update-deps.sh (hunk 2 above)
Comment (skonto) [reply hack/update-deps.sh:18]: We use the dep in tests too: https://github.com/openshift-knative/serverless-operator/blob/main/test/upgrade/upgrade_test.go#L27, so let's remove that in another PR.

**Review by pierDipi — APPROVED** [review]
/lgtm

File: hack/update-deps.sh (hunk 2 above)
Comment (pierDipi) [reply hack/update-deps.sh:18]: that code dependency is transitively imported by eventing-kafka-broker, so we will keep it but it's not gonna be enforced anymore

Comment (skonto) [reply hack/update-deps.sh:18]: I can remove the one line if that is the goal here. It has no effect, it does not touch go.mod. Is that what you are describing?
